# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../mc/slang.patch
# Copyright (C) 2008 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

sed -i -e 's,_SLsys_\(input_pending\|getkey\),SLang_\1,g' ./src/slint.c
--- ./src/slint.c.orig	2008-08-19 17:16:48.000000000 -0400
+++ ./src/slint.c	2008-08-19 17:42:32.000000000 -0400
@@ -75,8 +75,8 @@
 extern unsigned int SLang_Input_Buffer_Len;
 extern unsigned char SLang_Input_Buffer [];
 #if SLANG_VERSION >= 10000
-extern unsigned int _SLsys_getkey (void);
-extern int _SLsys_input_pending (int);
+extern unsigned int SLang_getkey (void);
+extern int SLang_input_pending (int);
 #else
 extern unsigned int SLsys_getkey (void);
 extern int SLsys_input_pending (int);
@@ -98,7 +98,7 @@
 	return(ch);
      }
 #if SLANG_VERSION >= 10000
-   else return(_SLsys_getkey ());
+   else return(SLang_getkey ());
 #else
    else return(SLsys_getkey());
 #endif
@@ -110,7 +110,7 @@
    unsigned char c;
    if (SLang_Input_Buffer_Len) return (int) SLang_Input_Buffer_Len;
 #if SLANG_VERSION >= 10000  
-   n = _SLsys_input_pending (tsecs);
+   n = SLang_input_pending (tsecs);
 #else
    n = SLsys_input_pending (tsecs);
 #endif
